packages/main/src/SplitButton.ts

```diff
@@ -347,22 +408,119 @@ class SplitButton extends UI5Element {
 
 	_fireArrowClick(e?: Event) {
 		e?.stopPropagation();
+
 		this.fireEvent("arrow-click");
 	}
 
 	_textButtonRelease() {
 		this._textButtonActive = false;
 		this._textButtonIcon = this.textButton && this.activeIcon !== "" && (this._textButtonActive) && !this._shiftOrEscapePressed ? this.activeIcon : this.icon;
-		this._setTabIndexValue();
+		this._tabIndex = "-1";
+	}
+
+	_arrowButtonPress(e: MouseEvent) {
+		e.preventDefault();
+		this.arrowButton!.focus();
+
+		this._tabIndex = "-1";
+	}
+
+	_arrowButtonRelease(e: MouseEvent) {
+		e.preventDefault();
+
+		this._tabIndex = "-1";
 	}
 
 	_setTabIndexValue() {
-		const textButton = this.textButton,
-			arrowButton = this.arrowButton,
-			buttonsAction = (textButton && (textButton.focused || textButton.active))
-						 || (arrowButton && (arrowButton.focused || arrowButton.active));
+		this._tabIndex = this.disabled ? "-1" : "0";
+
+		if (this._tabIndex === "-1" && (this.textButton?.focused || this.arrowButton?.focused)) {
+			this._tabIndex = "0";
+		}
+	}
 
-		this._tabIndex = this.disabled || buttonsAction ? "-1" : "0";
+	_setEffectiveActiveState() {
+		this._activeArrowButton = !!this.activeArrowButton;
```

Comment:
1. The values are boolean, no need of double "!!"

2. you don't need to update this._activeArrowButton onAfterRendering and cause sequential re-rendering of the component this._activeArrowButton = !!this.activeArrowButton
You just need the effective active state of the arrow button, giving priority to the public property and fallback to the internally maintained state, e.g:

get effectiveActiveArrowButton() {
		return this.activeArrowButton || this._activeArrowButton;
	}
and then using it the template, e.g
?active="{{effectiveActiveArrowButton}}"